utils/src/test/scala/za/co/absa/enceladus/utils/types/FormatSuite.scala

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2018-${today.year} ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.utils.types
+
+import java.security.InvalidParameterException
+import org.apache.spark.sql.types.{DateType, DoubleType, TimestampType}
+import org.scalatest.FunSuite
+
+class FormatSuite extends FunSuite {
+
+  test("Format class for timestamp") {
+    val pattern: String = "yyyy~mm~dd_HH.mm.ss"
+    val format: Format = new Format(Some(pattern), Some(TimestampType))
+    assert(!format.isDefault)
+    assert(format.get == pattern)
+    assert(format.getOrElse("foo") == pattern)
+    assert(!format.isEpoch)
+    val expectedMessage = s"'${format.get}' is not an epoch format"
+    try{
+      Format.epochFactor(format)
+      assert(false, "Should not be reached");
+    } catch {
+      case e: InvalidParameterException => assert(e.getMessage == expectedMessage);
```

Comment:
There is a better way to catch the errors in tests with `intercept`

```
val caught = intercept[InvalidParameterException] {
      Format.epochFactor(format)
    }
assert(caught.getMessage == expectedMessage)
```

There are more occurrences, where this can be applied.

Reply:
Thanks will redo